qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/IndexingIT.java

```diff
@@ -147,6 +234,88 @@ public void testIndexing() throws IOException {
         }
     }
 
+
+    /**
+     * This test verifies that during rolling upgrades the segment replication does not break when replica shards can
+     * be running on older codec versions.
+     *
+     * @throws Exception
+     */
+    public void testIndexingWithSegRep() throws Exception {
+        if (UPGRADE_FROM_VERSION.before(Version.V_2_4_0)) {
```

Comment:
Segrep release was in 2.7, there are no codec checks before that, do we still want to run this from 2.4? I'm assuming on previous versions (<2.7), primary will just send over segments on higher codec that replicas won't be able to read in mixed state and shard will be failed.

Reply:
The intention of the tests is to validate backward compatibility with older versions where feature was supported. This is to cover use cases where end uses indeed upgrade from 2.4 -> 2.7+ versions. Since codec check is not present prior to 2.7 versions, this test catches the issue. I feel we should strive towards writing tests according to use cases vs known code bugs.